src/results/report/handlebars.rs

```diff
@@ -46,31 +46,28 @@ impl HandlebarsReport {
             let dir_entry = dir_entry?;
             if let Some(ext) = dir_entry.path().extension() {
                 if ext == "hbs" {
-                    handlebars.register_template_file(dir_entry.path()
-                                                    .file_stem()
-                                                    .ok_or_else(|| {
-                                                        Error::TemplateName("template files must \
-                                                                             have a file name"
-                                                            .to_owned())
-                                                    })?
-                                                    .to_str()
-                                                    .ok_or_else(|| {
-                                                        Error::TemplateName("template names must \
-                                                                             be unicode"
-                                                            .to_owned())
-                                                    })?,
-                                                dir_entry.path())?;
+                    let path = dir_entry.path();
+                    let template_file = path
+                        .file_stem()
+                        .ok_or(ErrorKind::TemplateName("template files must have a file name".into()))
```

Comment:
The problem with `ok_or()` is that it will run the code even if it does not get there. Something like `or_else()` could be better, anything that requires a closure, basically.

Love how clean this code looks now, though.